src/paperbot/api/routes/runbook.py

```
@@ -865,6 +1202,96 @@ async def _run_smoke_async(run_id: str, workdir: Path, body: SmokeRequest) -> No
             pass
 
 
+async def _run_step_async(
+    run_id: str, step_name: str, workdir: Path, commands: List[str], body: StepRequest
+) -> None:
+    """Generic async step runner - follows _run_smoke_async pattern but parameterized by step_name."""
+    logger = get_execution_logger()
```

Ruff (0.15.1)

[error] Undefined name `get_execution_logger`

(F821)